### DISCUSSION TITLE
Typescript not generating allure results when no reporter provided. With mocha-allure-reporter steps are missing

### DISCUSSION THREAD

**Discussion by tlfulek** [Shelex/cypress-allure-plugin#218] [Q&A]
Hi, 
I have a typescript Cypress. 
When I use a command only with **--env allure=true** the folder allure-results is not created. 
I've read somewhere, that I should additionally provide **--reporter mocha-allure-reporter**.  When I use it allure-results is created, but it contains xml files, and the results contain only describe() and it(), the entire body of report such as steps are missing.

Could You give me an advise on which reporter to choose to make it work properly?

**Comment by jeremie-val** [comment]
Is it linked to this cypress issue ? 
https://github.com/cypress-io/cypress/issues/27113

**Comment by Shelex** [comment]
1) Yes, that may be caused by #216 
2) `--reporter mocha-allure-reporter` is absolutely separate library - https://github.com/allure-framework/allure-mocha which works with `mocha` as a reporter, thus has no access to cypress runtime. It produces xml because it was built for allure v1. As it works just with mocha it has no idea about cypress and how it works in the runtime, so you may get some basic output and no runtime information. In addition, cypress works with mocha in some specific way, so edge-cases are likely to happen when tests in report will be structured in a weird way.

> which reporter to choose

That is for you to decide. There is also a `mochawesome` reporter, which gives quite nice page with test info. Probably there are many more options which I am not aware of though.
